src/Core/Addon/Theme/Theme.php

```diff
@@ -75,4 +86,20 @@ public function onReset()
     {
         return true;
     }
+
+    public function setPageLayouts(array $layouts)
+    {
+        $this->settings['page_layouts'] = $layouts;
+    }
+
+    public function getLayoutForPage($page)
+    {
+        $layout_name = $this->theme_settings['default_layout'];
+        if ($this->settings['page_layouts'][$page]) {
```

Comment:
If a page is added to PrestaShop, for instance by a module, then the layout for this page will not have been set and this will produce a notice even though using the default layout is perfectly acceptable in this case, so I think you should check if `$this->settings['page_layouts'][$page]` is set.